### PR TITLE
Switching to the new steps/commands layout

### PR DESCRIPTION
**Pull request by jer**
- Array of commands instead of hash of key/value pairs
- Also cleaned up some docs and logs

### CODE REVIEW

**Comment by stjohnjohnson** [comment]
Lol, you committed as "Author: sd-buildbot <dev-null@screwdriver.cd>"

File: screwdriver/screwdriver_test.go [hunk 1]
```diff
@@ -186,9 +186,12 @@ func TestPipelineFromYaml(t *testing.T) {
 	    "main": [
 	      {
 	        "image": "node:4",
-	        "steps": {
-	          "install": "npm install"
-	        },
+	        "commands": [
+	          {
+						  "name": "install",
```
Comment (stjohnjohnson) [inline screwdriver/screwdriver_test.go]: Spacing?

**Comment by stjohnjohnson** [comment]
LGTM except for the author and the spacing

**Comment by jer** [comment]
Good catches. Guess I'll start using Docker to test this thing :P

**Comment by Filbird** [comment]
👍

**Comment by nkatzman** [comment]
👍

**Pull request merged**